webmunk-ext/src/background/worker.js

```diff
@@ -6,10 +6,64 @@ import { wmSessionMgr } from '@webmunk/utils';
 import "@webmunk/extension-ads/worker.js";
 import "@webmunk/cookies-scrapper-module/worker";
 
-const appMgr =  {
-  initialize:function(){
-    messenger?.addReceiver('appMgr', this);
-  }
+const appMgr = {
+  surveys: [],
+  completedSurveys: [],
+
+  initialize: async function () {
+    messenger?.addReceiver("appMgr", this);
+
+    await this.initSurveys();
+
+    chrome.tabs.onUpdated.addListener((tabId, changeInfo, tab) => {
+      if (changeInfo.status === "complete") {
```

Comment:
move this inside surveyCompleteListener to keep all listener logic in one method